### PR TITLE
New option prefix for match-document-filename rule

### LINKED ISSUE #1139: new option `prefix` for `match-document-filename` rule
in GraphQL Hive we have a convention of naming operations -> prefix them with `mutation.*`, `query.* `https://github.com/kamilkisiela/graphql-hive/tree/main/packages/web/app/src/graphql

Current rule support only `suffix` option https://github.com/B2o5T/graphql-eslint/blob/master/packages/plugin/src/rules/match-document-filename.ts#L23-L26

cc @dotansimha

### PR DESCRIPTION
**Pull request by TuvalSimha**
Related #1139

### CODE REVIEW

File: packages/plugin/tests/match-document-filename.spec.ts [hunk 1]
```diff
@@ -40,13 +40,71 @@ ruleTester.runGraphQLTests<[MatchDocumentFilenameRuleConfig]>('match-document-fi
       code: 'query SAMEAsOperation { foo }',
       options: [{ query: 'matchDocumentStyle' }],
     },
+    {
```
Comment (dotansimha) [inline packages/plugin/tests/match-document-filename.spec.ts:43]: can you add tests for valid + invalid with both prefix and suffix?

Comment (dotansimha) [inline packages/plugin/tests/match-document-filename.spec.ts:43]: Also maybe test more things like fragment/query/subscription?

**Review by dotansimha — COMMENTED** [review]
- [ ] add changeset
- [x] add missing tests
- [x] make sure CI passes
- [ ] test in another project (alpha release)

File: .changeset/light-lizards-know.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@graphql-eslint/eslint-plugin': major
```
Comment (dotansimha) [inline .changeset/light-lizards-know.md:2]: change to `minor`. @TuvalSimha

**Review by dotansimha — CHANGES_REQUESTED** [review]

File: .changeset/light-lizards-know.md (hunk 2 above)
Comment (TuvalSimha) [reply .changeset/light-lizards-know.md]: Done

**Review by dotansimha — APPROVED** [review]

**Review by B2o5T — APPROVED** [review]

**Pull request merged**